pkg/kubelet/volumes.go

```diff
@@ -264,23 +248,22 @@ func (kl *Kubelet) getPodVolumesFromDisk() map[string]cleanerTuple {
 }
 
 func (kl *Kubelet) newVolumeMounterFromPlugins(spec *volume.Spec, pod *api.Pod, opts volume.VolumeOptions) (volume.Mounter, error) {
+	// returns plugin if only 1 match is found or an error if 0 or more than 1 plugins are matched
 	plugin, err := kl.volumePluginMgr.FindPluginBySpec(spec)
 	if err != nil {
 		return nil, fmt.Errorf("can't use volume plugins for %s: %v", spec.Name(), err)
 	}
-	if plugin == nil {
-		// Not found but not an error
-		return nil, nil
-	}
 	physicalMounter, err := plugin.NewMounter(spec, pod, opts)
 	if err != nil {
-		return nil, fmt.Errorf("failed to instantiate volume physicalMounter for %s: %v", spec.Name(), err)
+		return nil, fmt.Errorf("failed to instantiate physicalMounter for volume: %s using plugin: %s with a root cause: %v", spec.Name(), plugin.Name(), err)
 	}
 	glog.V(10).Infof("Used volume plugin %q to mount %s", plugin.Name(), spec.Name())
 	return physicalMounter, nil
 }
 
 func (kl *Kubelet) newVolumeAttacherFromPlugins(spec *volume.Spec, pod *api.Pod, opts volume.VolumeOptions) (volume.Attacher, error) {
+	// FindAttachablePluginBySpec can return an Attacher or an error or a nil detacher and error
+	// as not all volumes will have an attacher/detacher
 	plugin, err := kl.volumePluginMgr.FindAttachablePluginBySpec(spec)
```

Comment:
@pmorie - I removed from here as the def of FindAttachablePluginBySpec already has a good godoc